packages/medusa/src/utils/format-registration-name.js

```diff
@@ -0,0 +1,42 @@
+import path from "path"
+
+/**
+ * Formats a filename into the correct container resolution name.
+ * Names are camelCase formatted and namespaced by the folder i.e:
+ * models/example-person -> examplePersonModel
+ * @param {string} fn - the full path of the file
+ * @return {string} the formatted name
+ */
+function formatRegistrationName(fn) {
+  const parsed = path.parse(fn)
+  const parsedDir = path.parse(parsed.dir)
+
+  const rawname = parsed.name
+  let namespace = parsedDir.name
+  if (namespace.startsWith("__")) {
+    const parsedCoreDir = path.parse(parsedDir.dir)
+    namespace = parsedCoreDir.name
+  }
+
+  switch (namespace) {
+    case "repositories":
+      namespace = "repositorys"
```

Comment:
Consider adding a tiny comment explaining the need for renaming, since this is not obvious